GoogleForCreators/web-stories-wp
# Dashboard: Move focus to left nav on toggle button click

## Summary
Controls focus when dashboard toggle menu button is clicked and left rail is expanded.

## Relevant Technical Choices
- In Storybook - wraps `PageHeader` in `NavProvider` to make demoing toggle easier 
- Swaps `visibility` for `opacity` on view toggle for left rail so that `.focus()` can get applied. 
- Adds a `useLayoutEffect` hook to focus left rail when toggle occurs and menu expands
- Adds aria role 'navigation' to left rail 
- Adds list elements to left rail navigation links, helps with tabbing through content and is more semantic. 

## User-facing changes
- Now when you have the smaller viewport hamburger menu that expands the left nav and you click or 'enter' into it, you will have your focus moved to the left nav and should be able to tab through content. 

![tabbing nav](https://user-images.githubusercontent.com/10720454/82253490-f2771c80-9905-11ea-84e3-be3c31b9f305.gif)

## Testing Instructions
- Key through dashboard and see that tabbing through the navigation works as you'd expect
- Verify that clicking also still works 
- Verify that layout is still correct (should have no changes other than focusable elements) 
- See that [storybook](http://localhost:55141/?path=/story/dashboard-components-pageheading--default) `Dashboard/Components/pageHeader` can now toggle (if viewport is < 1340px)

Note, you might have to tweak your browser a little bit as firefox and safari handle keying through browsers differently than chrome out of the box. I found I had to add `accessibility.tabfocus` like in here: https://stackoverflow.com/questions/11704828/how-to-allow-keyboard-focus-of-links-in-firefox to get firefox to work for me. And in Safari you need to open your preferences, navigate to advanced and check 'press tab to highlight each item on a webpage'. 

---



Fixes #1786
